include/igl/embree/EmbreeDevice.h

```diff
@@ -0,0 +1,61 @@
+// This file is part of libigl, a simple c++ geometry processing library.
+//
+// Copyright (C) 2020 Vladimir Fonov <vladimir.fonov@gmail.com>
+//               2013 Alec Jacobson <alecjacobson@gmail.com>
+//               2014 Christian Schüller <schuellchr@gmail.com>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can
+// obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+#ifndef IGL_EMBREE_EMBREE_DEVICE_H
+#define IGL_EMBREE_EMBREE_DEVICE_H
+#include <embree3/rtcore.h>
+#include <iostream>
+
+namespace igl
+{
+  namespace embree
+  {
+     // keep track of embree device
+     class EmbreeDevice
+     {
+       private:
+        static RTCDevice g_device;
+        static int       g_cntr;
```

Comment:
What's the purpose of this counter? All libigl function can share the same device for all we know.

Reply:
to be able to release RTCDevice and destroy completely, when nobody is using it. For example, it is possible to initialize it with different parameters (for whatever reasons).

Reply:
I'm not sure what are the advantages of being able to release the device concretely. In any case, due to the header-only nature of libigl those global variables should be encapsulated in a singleton "DeviceManager" class for linkage reasons.